app/actions/local/channel.ts

```diff
@@ -113,343 +108,328 @@ export async function switchToChannel(serverUrl: string, channelId: string, team
                 console.log('channel switch to', channel?.displayName, channelId, (Date.now() - dt), 'ms'); //eslint-disable-line
             }
         }
+
+        return {models};
     } catch (error) {
+        // eslint-disable-next-line no-console
+        console.log('Failed to switch to channelId', channelId, 'teamId', teamId, 'error', error);
         return {error};
     }
-
-    return {models};
 }
 
 export async function removeCurrentUserFromChannel(serverUrl: string, channelId: string, prepareRecordsOnly = false) {
-    const serverDatabase = DatabaseManager.serverDatabases[serverUrl];
-    if (!serverDatabase) {
-        return {error: `${serverUrl} database not found`};
-    }
-
-    const {operator, database} = serverDatabase;
-
-    const models: Model[] = [];
-    const myChannel = await getMyChannel(database, channelId);
-    if (myChannel) {
-        const channel = await myChannel.channel.fetch();
-        if (!channel) {
-            return {error: 'myChannel present but no channel on the database'};
-        }
-        models.push(...await prepareDeleteChannel(channel));
-        let teamId = channel.teamId;
-        if (teamId) {
-            teamId = await getCurrentTeamId(database);
-        }
+    try {
+        const {operator, database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+
+        const models: Model[] = [];
+        const myChannel = await getMyChannel(database, channelId);
+        if (myChannel) {
+            const channel = await myChannel.channel.fetch();
+            if (!channel) {
+                throw new Error('myChannel present but no channel on the database');
+            }
+            models.push(...await prepareDeleteChannel(channel));
+            let teamId = channel.teamId;
+            if (teamId) {
+                teamId = await getCurrentTeamId(database);
+            }
 
-        // We update the history ASAP to avoid clashes with channel switch.
-        await removeChannelFromTeamHistory(operator, teamId, channel.id, false);
+            // We update the history ASAP to avoid clashes with channel switch.
+            await removeChannelFromTeamHistory(operator, teamId, channel.id, false);
 
-        if (models.length && !prepareRecordsOnly) {
-            try {
-                await operator.batchRecords(models);
-            } catch {
-                // eslint-disable-next-line no-console
-                console.log('FAILED TO BATCH CHANGES FOR REMOVE USER FROM CHANNEL');
+            if (models.length && !prepareRecordsOnly) {
+                try {
```

Comment:
Probably not a big deal, but do we want nested try catch? And shouldn't we be returning the error in this case?